### ISSUE TITLE
Auto-protect formula cells

### ISSUE THREAD

**Issue by etowle** [etowle/1822#24]
After a new sheet is created, formula cells should be automatically protected.

**Comment by etowle** [comment]
Cell headers should also be protected.

**Comment by etowle** [comment]
I think the best way to do this is to manually add the desired protections to the base/SR1 sheet, then copy protections each time a new sheet is created. Thus, I protected most of the cell header and formula cells in each template's base/SR1 sheet.

A big downside is that setting protections is expensive, and a protected range must be a connected rectangle. For this reason, I tried to cover only the necessary formula and header cells using as few protections as necessary. This still requires ~25 separate ranges for each template. Unfortunately, this adds quite a bit of time to the new sheet generation.

An alternative is to protect the _entire_ sheet, then unprotect the specific ranges that users can edit. I'm not sure how expensive it is to unprotect specific ranges, but the number of separate ranges to unprotect this way would be fewer than the number of ranges we currently protect. However, the sheet then has a lock icon ("protected") next to the sheet name, which is very off-putting. So, we will use individual range protections instead.
